aten/src/ATen/native/TensorShape.cpp

```diff
@@ -126,7 +126,7 @@ Tensor narrow(const Tensor& self, int64_t dim, int64_t start, int64_t length) {
   AT_CHECK(self.dim() > 0, "narrow() cannot be applied to a 0-dim tensor.");
   auto cur_size = self.size(dim);
   if (start < 0) {
-    AT_ERROR("start out of range");
+    start = cur_size + start;
```

Comment:
let's say start is a very, very small number (-1000000). Then start would be out of bounds after this line and we wouldn't be able to report the error here to the user.

You should try to call `maybe_wrap_dim` [(usage here)](https://github.com/pytorch/pytorch/blob/64834f6fb8055cf2d633353992288a8823b915ed/aten/src/ATen/native/TensorShape.cpp#L52) instead